server/plugin.go

```diff
@@ -158,16 +158,20 @@ func (p *Plugin) connectTeamsAppClient() error {
 	p.msteamsAppClientMutex.Lock()
 	defer p.msteamsAppClientMutex.Unlock()
 
-	if p.msteamsAppClient == nil {
-		msteamsAppClient := msteams.NewApp(
-			p.getConfiguration().TenantID,
-			p.getConfiguration().ClientID,
-			p.getConfiguration().ClientSecret,
-			&p.apiClient.Log,
-		)
-
-		p.msteamsAppClient = client_timerlayer.New(msteamsAppClient, p.GetMetrics())
+	// We don't currently support reconnecting with a new configuration: a plugin restart is
+	// required.
+	if p.msteamsAppClient != nil {
+		return nil
 	}
```

Comment:
Note that, before, we only ever created `p.msteamsAppClient` once: so calling `connectTeamsAppClient` multiple times just amounted to calling `p.msteamsAppClient.Connect()` multiple times. There's no reason to do that on a configuration change if we're not going to reflect an updated configuration.